### DISCUSSION TITLE
Flicker Issue

### DISCUSSION THREAD

**Discussion by iamtr0n** [riffnshred/nhl-led-scoreboard#238] [Hardware]
Riff, been following this for awhile and I’m impressed. I bit in and bought parts for it. I went with the Pi Zero and it works fine aside from the flicker. I built two, both identical and both same exact flicker issues. I tried to solder 4 & 18 and tried to change the hat to pws but the pi just hangs for a minute or two, restarts and just sits on the loading screen. I noticed the flicker is better but the new code slows it down and doesn’t officially boot past the load screen. I could wait 20 min and if it won’t do anything until I restart. The code goes back to normal and works fine once restarted. I was going to try with a RP3. Will that fix the flicker? Also what can I do to rectify the anti flicker mod not working correctly/holding. Any help is much appreciated.

**Comment by riffnshred** [comment]
The Pi Zero is heading out the door at this point and I think I just did that with the lastest update. Running on the Pi 3 will fix everything. Keep in mind that even with the Anti Flicker mod, there is still some Minor flickering that can happen. This is because some processes are running in the background. I find locking the frame rate using --led-limit-refresh=120 does a pretty good job but it affect the colors and the brightness a bit.  Do not use the --led-limit-refresh flag on the Pi Zero, it actually pushes the CPU a bit.

**Reply by iamtr0n** [reply]
Will try with the RP3! Thank you sir.

**Reply by riffnshred** [reply]
No problem bud. enjoy !!!

**Reply by falkyre** [reply]
Also on raspberry pis with multiple cpu cores, using the `isolcpus=3` in the /boot/cmdline.txt will help with some of the flicker.

**Reply by mij2** [reply]
Just commenting on here, with the latest version, I'm running a Pi Zero with the hardware mod. I installed the lite OS and removed some suggested from the rpi-rgb-led-matrix girhub documentation. I have next to zero flicker.

`sudo apt-get remove bluez bluez-firmware pi-bluetooth triggerhappy pigpio`

**Reply by angtang10** [reply]
@riffnshred Sorry, I am really new to this but got to the step of getting the scoreboard working! With the flicker, how do you lock in the frame rate? On the computer or on Putty? Also if that doesn't work @mij2 can you direct me to the documents that suggest things to remove? Thanks!

**Comment by xxxDrewedxxx** [comment]
Running the panel at less than 100% brightness (like 80%) and having a “good” (oversized) power-supply help out my flicker.  I also got less flicker when I went to separate power plugs (but same power supply) for the board and the hat/bonnet.  Might just be coincidence.

**Reply by rhazzed** [reply]
Interesting observation.  Wonder if  powering the pi 3 via the bonnet is contributing.

**Reply by riffnshred** [reply]
power can definitely affect the flickering but the bonnet and HAT use voltage regulators ( the two bigger chips on the board) to keep the voltage stable.
